lib/carto/named_maps/template.rb

```diff
@@ -183,7 +185,11 @@ def attributes_and_interactivity(layer_infowindow, layer_tooltip)
           }
         end
 
-        [attributes, interactivity.join(',')]
+        if layer_options[:interactivity]
+          interactivity = interactivity & layer_options[:interactivity].split(',')
```

Comment:
You can write this as

```ruby
interactivity &= layer_options[:interactivity].split(',')
```

Reply:
Ahh, didn't know arrays had the `&=` operator in addition to the `&` operator